### ISSUE TITLE
Will you support checking more than one metric

### ISSUE THREAD

**Issue by Southeast** [HariSekhon/nagios-plugins#166]
like this:
./check_redis_stats.pl -H 127.0.0.1 -P 1234 -s connected_clients,blocked_clients,used_memory -w 300:500,1:100,500M:2G

**Comment by HariSekhon** [comment]
@Southeast 

I have thought about this but it not only makes the code more complicated - the checks become less granular so in terms of root cause analysis it's a little slower to pinpoint the problem because you have to humanly scan more information.

Mainly it's just awkward to handle in terms of the code, plus the output variations of different single/multi fail scenarios, and code complexity is the enemy of stability and robustness which are really important in terms of enterprise monitoring code to try to be as bullet proof as possible.

Right now I don't have plans to extend support for checking more than one metric at a time and I wonder if this may be better handled via just having multiple checks or a check multi style wrapper plugin which would be a more generic solution than doing this on a plugin-by-plugin basis.

The time taken for this will also distract from releasing a lot of other useful plugins for new technologies so it won't be as high a priority as completing coverage of new or not-yet-covered technologies.

I'm open to feedback on this.

Thanks

Hari

**Comment by Southeast** [comment]
Sir @HariSekhon 
Thanks for replying!
I found perl compilation is very expensive when doing a check via perl. I have 1000 redis nodes,  Doing active checks all on a single icinga master server resulted in extremely high CPU load(150-200).
 I have resolve the CPU load problem by usingi distributed monitoring(1 master, 16 satellites), but I still can't afford  having multiple checks for multi metrics. I mean, in the case of big service clusters, checking multi metrics in one check seems important. 

Thanks!

**Comment by HariSekhon** [comment]
Yes I appreciate this is a trade off of scale and performance vs code.

I was going to say that distributed monitoring solution is generally the way to go to scale as it's easier to just use more computer power than more limited manpower.

That being said it is open source so you're free to modify it to suit your needs and submit a patch request if you need to... and I see it's pretty much this one check in this case.

I'll leave the ticket open as something for the backlog and if do get round to it before me (you almost certainly will as I'm already backlogged with no bandwidth this year) then you can reference this ticket in the patch pull request.

**Comment by Southeast** [comment]
I understand, thank you.
I'll try to learn perl and do something on it. 
Thanks again, and I'll close the issue.
